packages/front-end/components/Features/ConditionInput.tsx

```diff
@@ -107,6 +121,15 @@ export default function ConditionInput(props: Props) {
           {conds.map(({ field, operator, value }, i) => {
             const attribute = attributes.get(field);
 
+            const savedGroupOptions = savedGroups.filter((g) => {
```

Comment:
`.filter` should return a boolean.  I think what you're trying to do is filter and then map.

```ts
const savedGroupOptions = savedGroups
  // First, limit to groups with the correct attribute
  .filter((g) => g.attributeKey === field)
  // Then, transform into the select option format
  .map((g) => ({
    label: g.groupName,
    value: g.id,
  }));
```

Then, you can use the variable below when passing into `<SelectField options={savedGroupOptions}>`

Reply:
mmmmm.... that's much cleaner. I always tend to forget about chaining.